crates/cairo-coverage-core/src/build/filter/ignore_matcher.rs

```diff
@@ -0,0 +1,69 @@
+use crate::build::filter::statement_category_filter::VIRTUAL_FILE_REGEX;
+use anyhow::{Error, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use ignore::gitignore::Gitignore;
+use ignore::Match;
+
+const CAIRO_COVERAGE_IGNORE: &str = ".cairo-coverage-ignore";
+
+/// Create a new instance of the [`CairoCoverageIgnoreMatcher`] that will be based on the [`CAIRO_COVERAGE_IGNORE`] file.
+pub fn build(path: &Utf8Path) -> Result<CairoCoverageIgnoreMatcher> {
+    let ignore_matcher = find_ignore_file(path)
+        .map(Gitignore::new)
+        .map(|(ignore, error)| {
+            if let Some(error) = error {
+                Err(Error::from(error))
+            } else {
+                Ok(ignore)
+            }
+        })
+        .transpose()?
+        .unwrap_or_else(Gitignore::empty);
+
+    Ok(CairoCoverageIgnoreMatcher(ignore_matcher))
+}
+
+/// Search for a [`CAIRO_COVERAGE_IGNORE`] file in the given directory.
+fn find_ignore_file(dir: &Utf8Path) -> Option<Utf8PathBuf> {
+    let candidate = dir.join(CAIRO_COVERAGE_IGNORE);
+    candidate.is_file().then_some(candidate)
+}
+
+pub struct CairoCoverageIgnoreMatcher(Gitignore);
+
+impl CairoCoverageIgnoreMatcher {
+    /// Check if the given path is ignored by the [`CAIRO_COVERAGE_IGNORE`] file.
+    pub fn is_ignored(&self, path: &str) -> bool {
+        let path: Utf8PathBuf = VIRTUAL_FILE_REGEX.replace_all(path, "").to_string().into();
+        let result = self.0.matched(&path, path.is_dir());
+        matches!(result, Match::Ignore(_))
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
nit: could also use `test_finds_ignore_file_in_parent_directory` testcase to make sure a file in the workspace root directory is indeed found properly